core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -586,6 +612,16 @@ protected void writeSnapshotList(List<SnapshotId> snapshots) throws IOException
      * @throws IOException I/O errors
      */
     protected List<SnapshotId> readSnapshotList() throws IOException {
+        try {
+            if (snapshotsBlobContainer.blobExists(SNAPSHOTS_FILE) == false) {
```

Comment:
why do we need to add this here? Can you explain please? Also I'm wondering why blobExists is not supported by URLRepository (Shouldn't that just correspond to a HEAD request to the resource?).